starknet_py/net/account/account.py

```diff
@@ -79,26 +83,40 @@ def __init__(
         :param key_pair: Key pair that will be used to create a default `Signer`.
         :param chain: ChainId of the chain used to create the default signer.
         """
-        if chain is None and signer is None:
-            raise ValueError("One of chain or signer must be provided.")
-
         self._address = parse_address(address)
         self._client = client
 
-        if signer is not None and key_pair is not None:
-            raise ValueError("Arguments signer and key_pair are mutually exclusive.")
+        if signer is not None:
+            self._assert_no_key_pair_chain(key_pair, chain)
 
-        if signer is None:
-            if key_pair is None:
-                raise ValueError(
-                    "Either a signer or a key_pair must be provided in Account constructor."
-                )
+        self.signer = signer or self._create_signer(key_pair, chain)
```

Comment:
What if we rewrote this to something like

```py
if signer is not None:
  self._assert_no_key_pair_chain(key_pair, chain)
  self.signer = signer
else:
  self.signer = self._create_signer(key_pair, chain)
```
I think it makes it easier to follow what happens in cases signer is and isn't provided.